libs/ktxreader/include/ktxreader/Ktx2Reader.h

```diff
@@ -85,16 +93,107 @@ class Ktx2Reader {
+             *
+             * The texture is available immediately, but does not have its miplevels ready until
+             * after uploadImages(). The caller has ownership over this texture and is responsible
+             * for freeing it.
+             */
+            Texture* getTexture() const noexcept;
+
+            /**
+             * Loads mipmaps from the KTX2 file and transcodes them to the resolved format.
+             *
+             * This is typically called from a background thread.
+             */
+            Result doTranscoding();
+
+            /**
+             * Uploads all mipmaps to the texture.
+             *
+             * This calls Texture::setImage() and potentially generates missing mipmaps.
+             * If transcoding has not yet completed, returns INCOMPLETE.
+             *
+             * NOTE: This can be called only from the foreground thread.
+             */
+            void uploadImages();
+
+        protected:
+            Async() noexcept = default;
+            ~Async() = default;
+
+        public:
+            Async(Async const&) = delete;
+            Async(Async&&) = delete;
+            Async& operator=(Async const&) = delete;
+            Async& operator=(Async&&) = delete;
+
+            friend class Ktx2Reader;
+        };
+
+        /**
+         * Creates a texture without starting the transcode process.
+         *
+         * This method is an alternative to load() that allows users to populate mipmap levels
+         * asynchronously. The texture object however is still created synchronously.
+         *
+         * - For a usage example, see the documentation for the Async object.
+         * - Creates a copy of the given buffer, allowing clients to free it immediately.
+         * - Returns null if none of the requested formats can be extracted from the data.
+         *
+         * This method iterates through the requested format list, checking each one against the
+         * platform's capabilities and its availability from the transcoder. When a suitable format
+         * is determined, it then performs lossless decompression (zstd) before transcoding the data
+         * into the final format.
+         *
+         * The transfer function specified here is used in two ways:
+         *   1) It is checked against the transfer function that was specified as metadata
+         *      in the KTX2 blob. If they do not match, this method fails.
+         *   2) It is used as a filter when determining the final internal format.
+         */
+        Async* asyncCreate(const uint8_t* data, size_t size, TransferFunction transfer);
+
+        /**
+         * Frees the given async object if it is non-null.
+         *
+         * This frees the original source data (i.e. the raw content of the KTX2 file) but does not
+         * free the associated Texture object. This can be after transcoding has finished.
+         */
+        void asyncDestroy(Async* async);
```

Comment:
I like to pass a `Async**` in the destroy functions, so that the passed pointer can be set to nullptr. This makes it impossible (or hard) to use after calling destroy:
e.g.:
```
  asyncDestroy(&mAsync);
```

Reply:
Will do.  I noticed that most of our `Engine::destroy` methods don't do this, but I'm guessing that's historical.